frontend/apps/reader/modules/readerdevicestatus.lua

```diff
@@ -0,0 +1,123 @@
+local ButtonDialogTitle = require("ui/widget/buttondialogtitle")
+local Device = require("device")
+local Font = require("ui/font")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local Screen = Device.screen
+local UIManager = require("ui/uimanager")
+local powerd = Device:getPowerDevice()
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local ReaderDeviceStatus = InputContainer:new{
+}
+
+function ReaderDeviceStatus:init()
+    self.battery_watchable = powerd:getCapacity() > 0 or powerd:isCharging()
+    self.check_low_battery = function()
+        local threshold = G_reader_settings:readSetting("low_battery_threshold") or 20
+        local battery_capacity = powerd:getCapacity()
+        if powerd:getDissmisBatteryStatus() ~= true and battery_capacity <= threshold then
+            local low_battery_info
+            low_battery_info = ButtonDialogTitle:new {
+                modal = true,
```

Comment:
Is that `modal=true` a left-over from the previous version that you replaced with `dismissal`, or a real use of the `modal` option to widgets (checked by UIManager for some obscure window stack management)?

Reply:
This is required. Without `modal=true` this window can be hidden behind filemanager or reader on startup.